packages/core/src/internal/decorators/i18n.ts

```diff
@@ -96,3 +110,45 @@ export function i18n() {
       : standardI18n(descriptor, protoOrDescriptor);
   };
 }
+
+export function getI18nValues(values: Record<string, unknown>, component: I18nElement): Record<string, unknown> {
+  if (isNilOrEmpty(values)) {
+    let returnVal = {};
+
+    if (component.hasAttribute('cds-i18n')) {
+      const stringVal = component.getAttribute('cds-i18n') + '';
+      if (isNilOrEmpty(stringVal)) {
+        returnVal = {};
+      } else {
+        try {
+          returnVal = JSON.parse(stringVal);
+        } catch {
+          console.warn('Clarity i18n: Invalid JSON passed to cds-i18n');
+          returnVal = {};
+        }
+      }
+    }
+    return returnVal;
+  } else {
+    return values;
+  }
+}
+
+export function getI18nUpdateStrategy(
+  newKey: string,
+  oldKey: string,
+  newValues: Record<string, unknown>,
+  oldValues: Record<string, unknown>
+): { values?: object; key?: string; update: boolean } {
+  if (!isNilOrEmpty(newKey)) {
```

Comment:
Here's that strategy pattern. You send it your data. It returns with an object that tells you how the i18n should update/change.